### ISSUE TITLE
pydantic.utils.Representation deprecation

### ISSUE THREAD

**Issue by mukhery** [omni-us/jsonargparse#361]
## 🔧 Code refactor
https://github.com/omni-us/jsonargparse/blob/8356fdff453de4d044ba01d0119724481dcddc2b/jsonargparse/_common.py#L83-L85 calls `pydantic.utils.Representation` which appears to be deprecated, resulting in a warning message.
```
UserWarning: `pydantic.utils:Representation` has been removed. We are importing from `pydantic.v1.utils:Representation` instead.See the migration guide for more details: https://docs.pydantic.dev/latest/migration/
```

**Comment by mauvilsa** [comment]
Thank you for reporting! I did notice some time ago that unit tests fail with `pydantic>=2.0.0`, see [pyproject.toml#L90](https://github.com/omni-us/jsonargparse/blob/8356fdff453de4d044ba01d0119724481dcddc2b/pyproject.toml#L90). Unfortunately fixing this is not a small task, is it would make sense that jsonargparse works correctly for `pydantic<2.0.0` and `pydantic>=2.0.0`, and the pydantic breaking changes are significant.

@mukhery can you please provide a minimal code snippet that produces the warning described above. Also include the pydantic version you are using.

**Comment by mukhery** [comment]
```python
>>> pydantic.__version__
'2.1.1'
```
Will provide a code snippet soon.

**Comment by mauvilsa** [comment]
Ah, then I guess the deprecation warning is only shown with `pydantic>1`. I was wondering why I wasn't seeing this in the unit tests. If this is the case, then I don't need anymore a snippet to reproduce.

**Comment by mauvilsa** [comment]
This is fixed by pull request #389.
